### PR TITLE
CANS-1259 Adjust README and add Architecture section

### PR DESCRIPTION
**Pull request by TylerOSI**
## Description
Update CANS documentation for the front end!

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [ ] I ran all the linters for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,20 @@
 # CANS
 
-This README documents the steps necessary to get the
-application up and running.
+CANS stands for Child and Adolescent Needs and Strengths. It is an application built to capture the needs and strengths of children and adolescents through assessments. This repository contains the front end application which is explained in detail in the [architecture](#architecture) section.
```
Comment (TylerOSI) [inline README.md]: `CANS stands for Child and Adolescent Needs and Strengths` is a correct statement?

File: README.md [hunk 2]
```diff
@@ -106,6 +99,49 @@ yarn lint
 2. Hotfix branches should be protected in GitHub similar to master
 3. Hotfix pipelines in CANS should be changed to use new hotfix branches as a baseline
 
+# Architecture
+
+The CANS application front end has two parts Rails and React. Each part has its own libraries and design decisions which are covered in the respective sections. Rails serves the React application, proxies api calls, and stores session information using Redis. React is the front end application code that runs in the browser and makes requests to Rails for API calls. Rails is meant to be a simple proxy for API calls not a complex middle-man between the CANS API and the React application.
+
+## Rails
+
+[Rails](https://github.com/rails/rails) is an MVC Framework for developing web applications. Its usage in CANS is minimal because it primarily serves as session storage and a proxy for API calls. However, Rails is developed in a community with strong ties to [Test Driven Development](http://agiledata.org/essays/tdd.html) and automated testing methods which proves to be very valuable for development.
```
Comment (AK56) [inline README.md]: Apart from Rails serving as a proxy and session storage, it also houses our regression suite! Not sure, how much detail we would want to layout on our regression suite, in which case we can utilize resources/docs directory to place such documentation. Or we can just provide high level details around our regression suite here

File: README.md [hunk 3]
```diff
@@ -106,6 +99,49 @@ yarn lint
...
+
+### Primary Dependencies
+* [babel](https://babeljs.io/) for transpiling ES6 and JSX into JavaScript
+* [webpack](https://webpack.js.org/) for packaging all assets into packs
+* [component library](https://github.com/ca-cwds/design-system) a library for CWDS components used accross all CWDS services
+* [material UI](https://material-ui.com/) for custom components outside the component library (assessment form)
+* [axios](https://github.com/axios/axios) a web client for making API calls
+* [React](https://reactjs.org/) a library for developing web applications
+* [eslint](https://eslint.org/) a utility for checking that code follows cleanliness standards
+* [jest](https://jestjs.io/) a test framework for unit tests
```
Comment (AK56) [inline README.md]: Few other important libs we should mention: [reactstrap](https://github.com/reactstrap/reactstrap) - serving stateless React components for bootstrap 4, which we are also using in CANS, [downshift](https://github.com/downshift-js/downshift) - which fuels dropdown/autocomplete functionality to react components, and [recharts](https://github.com/recharts/recharts) - chart lib that we are using around Assessment Comparison domain

File: README.md (hunk 2 above)
Comment (TylerOSI) [reply README.md]: good call out ill add some comments on our regression suite.

File: README.md (hunk 3 above)
Comment (TylerOSI) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (LeonidMarushevskyi) [reply README.md]: You can mention that we are using Page Object Model and site-prism for regression testing.

**Comment by TylerOSI** [comment]
Jenkins test this please

**Comment by TylerOSI** [comment]
Jenkins please test this

**Comment by TylerOSI** [comment]
Jenkins test this please

**Comment by denys-davydov** [comment]
Retest this please

**Comment by edgibbs** [comment]
retest this please

**Pull request merged**